internal/service/route53resolver/firewall_domain_list_data_source.go

```diff
@@ -0,0 +1,90 @@
+package route53resolver
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/route53resolver"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+)
+
+func DataSourceResolverFirewallDomainList() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceResolverFirewallDomainListRead,
+
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"creation_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"creator_request_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"domain_count": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"managed_owner_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"modification_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"status_message": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceResolverFirewallDomainListRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*conns.AWSClient).Route53ResolverConn
+
+	input := &route53resolver.GetFirewallDomainListInput{
+		FirewallDomainListId: aws.String(d.Get("id").(string)),
+	}
+
+	output, err := conn.GetFirewallDomainList(input)
+
+	if err != nil {
+		return fmt.Errorf("error getting Route53 Firewall Domain List: %w", err)
+	}
+
+	if output == nil {
+		return fmt.Errorf("no  Route53 Firewall Domain List found matching criteria; try different search")
+	}
+
+	d.SetId(aws.StringValue(output.FirewallDomainList.Id))
```

Comment:
There are multiple attributes here that are not present in `ResourceFirewallDomainList`. I would recommend an additional PR to add these attributes to `ResourceFirewallDomainList`.